packages/app-config-writer/src/smartlinks-config/ui5-yaml.ts

```diff
@@ -0,0 +1,90 @@
+import { FileName, readUi5Yaml } from '@sap-ux/project-access';
+import type {
+    AbapDeployConfig,
+    CustomMiddleware,
+    FioriToolsProxyConfig,
+    FioriToolsServeStaticConfig,
+    FioriToolsServeStaticPath,
+    UI5Config
+} from '@sap-ux/ui5-config';
+import { t } from '../i18n';
+import type { TargetConfig } from '../types';
+import { DeployConfig } from '../types';
+
+/**
+ * Reads and returns target information from ui5-deploy.yaml, if existing.
+ *
+ * @param basePath - path to project root, where ui5-deploy.yaml is
+ * @returns {TargetConfig} target definition for deploy configuration
+ */
+export async function readUi5DeployConfigTarget(basePath: string): Promise<TargetConfig> {
+    const ui5DeployYaml = await readUi5Yaml(basePath, FileName.UI5DeployYaml);
+    const customTask = ui5DeployYaml.findCustomTask<AbapDeployConfig>(DeployConfig.DeployToAbap);
+    if (!customTask?.configuration?.target) {
+        throw Error(t('error.noTarget', { file: `(${FileName.UI5DeployYaml})` }));
+    }
+    const { target, ignoreCertError } = customTask?.configuration;
+    return { target, ignoreCertErrors: ignoreCertError };
+}
+
+/**
+ * Checks if 'fiori-tools-servestatic' configuration is already existing.
+ *
+ * @param existingPaths existing configuration paths under 'fiori-tools-servestatic'
+ * @returns 'fiori-tools-servestatic' configuration, if not existing yet
+ */
+const getFioriToolsServeStaticMiddlewareConfig = (
+    existingPaths: FioriToolsServeStaticPath[]
+): CustomMiddleware<FioriToolsServeStaticConfig> | undefined => {
+    const configPath: FioriToolsServeStaticPath = { path: '/appconfig', src: 'appconfig', fallthrough: false };
+    if (existingPaths.find((existing) => existing.path === configPath.path && existing.src === configPath.src)) {
+        return undefined;
+    }
+    const paths = [...existingPaths, configPath];
+    return {
+        name: 'fiori-tools-servestatic',
+        beforeMiddleware: 'fiori-tools-proxy',
+        configuration: { paths }
+    };
+};
+
+/**
+ * Reads and adds servestatic configuration to ui5/-local/-mock.yaml files.
+ *
+ * @param basePath - path to project root, where ui5.yaml is
+ * @param fs - mem-fs reference to be used for file access
+ * @param logger - logger
+ */
+export async function addUi5YamlServeStaticMiddleware(
+    basePath: string,
+    fs: Editor,
+    logger?: ToolsLogger
+): Promise<void> {
+    const ui5Yamls = [FileName.Ui5Yaml, FileName.Ui5MockYaml, FileName.Ui5LocalYaml];
+    for (const ui5Yaml of ui5Yamls) {
+        let ui5YamlConfig: UI5Config;
+        try {
+            ui5YamlConfig = await readUi5Yaml(basePath, ui5Yaml);
+        } catch (error) {
+            logger?.debug(`File ${ui5Yaml} not existing`);
+            continue;
+        }
+        if (!ui5YamlConfig.findCustomMiddleware<FioriToolsProxyConfig>(DeployConfig.FioriToolsProxy)) {
+            ui5YamlConfig.addFioriToolsProxydMiddleware({ ui5: {} });
```

Comment:
@zdravko-georgiev There was a dependency for fioritoolsproxy in servestatic, so in case there is no fioritoolsproxy we add it - or is that not expected?

Reply:
@johannes-kolbe This is not correct. The dependency to the `fiori-tools-proxy` is only for the case when reuse libs needs to be loaded from the local workspace. In this case we need to ensure that the `fiori-tools-servestatic` is called before the `fiori-tools-proxy`. For the case where there is no `fiori-tools-proxy`, then we can just add:
```
  - name: fiori-tools-servestatic
    afterMiddleware: compression
```

Reply:
Ok, thanks for the clarification! Updated PR with handling of non-existing `fiori-tools-proxy`